### PR TITLE
branches api using bitbucket api v2

### PR DESCRIPTION
**Pull request by falexandrou**
@hirakiuc Here is my work for the branches api

### CODE REVIEW

File: lib/tinybucket/api/branches_api.rb [hunk 1]
```diff
@@ -0,0 +1,48 @@
+# frozen_string_literal: true
+
+module Tinybucket
+  module Api
+    # Branches Api client
+    #
+    # @!attribute [rw] repo_owner
+    #   @return [String] repository owner name.
+    # @!attribute [rw] repo_slug
+    #   @return [String] repository slug. (about {https://developer.atlassian.com/bitbucket/api/2/reference/resource/repositories/%7Busername%7D/%7Brepo_slug%7D
```
Comment (hirakiuc) [inline lib/tinybucket/api/branches_api.rb]: Currently, this yard comment shows a raw URL in generated document. 😲 

You should add ` repo_slug})` after this line to create the link in yard document. 📘

File: lib/tinybucket/api/commits_api.rb [hunk 2]
```diff
@@ -7,16 +7,15 @@ module Api
     # @!attribute [rw] repo_owner
     #   @return [String] repository owner name.
     # @!attribute [rw] repo_slug
-    #   @return [String] repository slug. (about {https://confluence.atlassian.com/bitbucket/repositories-endpoint-423626330.html#repositoriesEndpoint-Overview
-    #     repo_slug})
+    #   @return [String] repository slug. (about {https://developer.atlassian.com/bitbucket/api/2/reference/resource/repositories/%7Busername%7D/%7Brepo_slug%7D
```
Comment (hirakiuc) [inline lib/tinybucket/api/commits_api.rb]: [Same comment](https://github.com/hirakiuc/tinybucket/pull/102/files#diff-56f8fa0baa8396b32d6afaf6af940686R10). 📘

File: lib/tinybucket/api/commits_api.rb [hunk 3]
```diff
@@ -49,6 +48,7 @@ def find(revision, options = {})
       end
 
       # Send 'POST a commit approval' request
+      # @see https://developer.atlassian.com/bitbucket/api/2/reference/resource/repositories/%7Busername%7D/%7Brepo_slug%7D/commit/%7Bnode%7D/approve#post
```
Comment (hirakiuc) [inline lib/tinybucket/api/commits_api.rb]: require linked text, like [here](https://github.com/hirakiuc/tinybucket/pull/102/files#diff-1649812b478d3dfd294d9add0dcaa50bR37). 📘

File: lib/tinybucket/api/commits_api.rb [hunk 4]
```diff
@@ -62,6 +62,7 @@ def approve(revision, options = {})
       end
 
       # Send 'DELETE a commit approval' request
+      # @see https://developer.atlassian.com/bitbucket/api/2/reference/resource/repositories/%7Busername%7D/%7Brepo_slug%7D/commit/%7Bnode%7D/approve#delete
```
Comment (hirakiuc) [inline lib/tinybucket/api/commits_api.rb]: require linked text, like [here](https://github.com/hirakiuc/tinybucket/pull/102/files#diff-1649812b478d3dfd294d9add0dcaa50bR37). 📘

File: lib/tinybucket/model/commit.rb [hunk 5]
```diff
@@ -4,7 +4,8 @@ module Tinybucket
   module Model
     # Commit Resource
     #
-    # @see https://confluence.atlassian.com/bitbucket/commits-or-commit-resource-389775478.html
+    # @see https://developer.atlassian.com/bitbucket/api/2/reference/resource/repositories/%7Busername%7D/%7Brepo_slug%7D/commit
+    #      https://developer.atlassian.com/bitbucket/api/2/reference/resource/repositories/%7Busername%7D/%7Brepo_slug%7D/commits
```
Comment (hirakiuc) [inline lib/tinybucket/model/commit.rb]: This yard comment shows the `commits URL` text which linked the `commit URL` in generated document.

How about this code if you want to create two links?

```diff
$ git diff lib/tinybucket/model/commit.rb
diff --git a/lib/tinybucket/model/commit.rb b/lib/tinybucket/model/commit.rb
index 8a251a3..fec4521 100644
--- a/lib/tinybucket/model/commit.rb
+++ b/lib/tinybucket/model/commit.rb
@@ -5,8 +5,9 @@ module Tinybucket
     # Commit Resource
     #
     # @see https://developer.atlassian.com/bitbucket/api/2/reference/resource/repositories/%7Busername%7D/%7Brepo_slug%7D/commit
-    #      https://developer.atlassian.com/bitbucket/api/2/reference/resource/repositories/%7Busername%7D/%7Brepo_slug%7D/commits
-    #   Commit Resource - Bitbucket Cloud REST API document
+    #   Commit resource - Bitbucket Cloud REST API document
+    # @see https://developer.atlassian.com/bitbucket/api/2/reference/resource/repositories/%7Busername%7D/%7Brepo_slug%7D/commits
+    #   Commits resource - Bitbucket Cloud REST API document
     #
     # @!attribute [rw] hash
     #   @return [String]
```

**Comment by hirakiuc** [comment]
At next Monday, I'll test by creating some codes to use new APIs. 💪

**Comment by hirakiuc** [comment]
Your code works good! 👍 

I tested with this sample code. 

```ruby
require 'tinybucket'
require 'logger'

logger = Logger.new($stdout).tap do |log|
  log.level = Logger::WARN
end

Tinybucket.configure do |config|
  config.logger = logger
  config.oauth_token = ENV['FEATURE_TEST_OAUTH_TOKEN']
  config.oauth_secret = ENV['FEATURE_TEST_OAUTH_SECRET']
end

bucket = Tinybucket.new
bucket.repos('hirakiuc').take(3).each do |repo|
  puts repo.name

  repo.branches.each do |branch|
    puts "  -> #{branch.name}"
  end

  begin
    p repo.branch('use_foreman')
  rescue Tinybucket::Error::NotFound
    puts '  => NotFound'
  end
end
```

NOTE: I always test with real Bitbucket REST API, because Bitbucket REST API document sometime  contains wrong information. :weary:

File: lib/tinybucket/model/branch.rb [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+    #   @return [Hash]
+    # @!attribute [rw] target
+    #   @return [Hash]
+    class Branch < Base
+      include Tinybucket::Model::Concerns::RepositoryKeys
+
+      acceptable_attributes :links, :type, :name, :repository, :target
+
+      def commits
+        commits_api.branch(name)
```
Comment (hirakiuc) [inline lib/tinybucket/model/branch.rb]: This code should use `Commits Resource` to be enumerable, just like [Repository#commits](https://github.com/hirakiuc/tinybucket/blob/master/lib/tinybucket/model/repository.rb#L109).

But this pull request has already many changes. 
So it is better to keep this code right now, and fixed in another issue, I think. 😄

**Comment by hirakiuc** [comment]
I've added some comments with requesting small changes. 💬 
Please check my comments. 😄

File: lib/tinybucket/model/repository.rb [hunk 7]
```diff
@@ -118,6 +118,39 @@ def commit(revision, options = {})
         commits_resource.find(revision, options)
       end
 
+      # Get the list of hooks on this repository.
+      #
+      # @param options [Hash]
+      # @return [Tinybucket::Resource::Hooks]
+      def hooks(options = {})
+        hooks_resource(options)
+      end
```
Comment (falexandrou) [inline lib/tinybucket/model/repository.rb]: Ooops this is by mistake. I will revisit my branch and remove the unnecessary code

**Comment by hirakiuc** [comment]
LGTM.

Thanks for your work! 👍

**Pull request merged**